File: src/models/try_winning_move.mjs
import calculateWinner from "./calculate_winner.mjs";

export default function tryWinningMove(currentBoard, emptySquaresIndexes, marker) {

    for (const emptySquare of emptySquaresIndexes) {
        const newSquares = currentBoard.slice()

        newSquares[emptySquare] = marker
    
        if (calculateWinner(newSquares)) {
            return emptySquare
        }
    }
    
    return null
}

File: api/try_winning_move.py
from calculate_winner import calculateWinner

def tryWinningMove(currentBoard, emptySquaresIndexes, marker):

    for emptySquare in emptySquaresIndexes:
        newSquares = currentBoard.copy()
        newSquares[emptySquare] = marker

        if calculateWinner(newSquares)["winner"]:
            return emptySquare
    
    return None